syft/pkg/catalog.go

```diff
@@ -81,15 +88,26 @@ func (c *Catalog) Add(p Package) {
 		id = p.ID()
 	}
 
+	if existing, exists := c.byID[id]; exists {
+		// there is already a package with this fingerprint merge the existing record with the new one
+		if err := existing.merge(p); err != nil {
+			log.Warnf("failed to merge packages: %+v", err)
+		}
+		return
+	}
+
```

Comment:
Do we need to update any of the maps on the catalog after the package has been merged or is `Add` the first time we see `p`?

Reply:
let me think on that. I think you might be right about needing to update indexes for paths.